corehq/apps/accounting/forms.py

```diff
@@ -160,6 +160,18 @@ def __init__(self, account, *args, **kwargs):
             )
         )
 
+    def clean_name(self):
+        existing_names = set(account.name for account in BillingAccount.objects.all())
```

Comment:
Not too big of a deal but could be more efficient to use `values` https://docs.djangoproject.com/en/1.7/ref/models/querysets/#django.db.models.query.QuerySet.values

`BillingAccount.objects.values('name')`